lib/src/components/NxTextLink/__tests__/NxTextLink.test.tsx

```diff
@@ -5,114 +5,106 @@
-    expect(getShallow()).toHaveProp('aria-disabled', false);
-    expect(getShallow({ disabled: true })).toHaveProp('aria-disabled', true);
+    expect(renderEl()).toHaveAttribute('aria-disabled', 'false');
+    expect(renderEl({ disabled: true })).toHaveAttribute('aria-disabled', 'true');
   });
 
   it('sets link role on the text link when the disabled prop is set', function() {
-    expect(getShallow()).not.toHaveProp('role', 'link');
-    expect(getShallow({ disabled: true })).toHaveProp('role', 'link');
+    expect(renderEl()).not.toHaveAttribute('role', 'link');
+    expect(renderEl({ disabled: true })).toHaveAttribute('role', 'link');
   });
 
   it('does not pass href value on the text link when the disabled prop is set', function() {
-    expect(getShallow({ href: '#/pages/Text%20Link' })).toHaveProp('href', '#/pages/Text%20Link');
-    expect(getShallow({ href: '#/pages/Text%20Link', disabled: true })).toHaveProp('href', undefined);
+    expect(renderEl({ href: '#/pages/Text%20Link' })?.getAttribute('href')).toContain('#/pages/Text%20Link');
+    expect(renderEl({ href: '#/pages/Text%20Link', disabled: true })?.getAttribute('href')).toBe(null);
   });
 
   it('sets the target to _blank if newTab is true, unless a different target is specified', function() {
-    expect(getShallow()).toHaveProp('target', '');
-    expect(getShallow({ newTab: true })).toHaveProp('target', '_blank');
+    expect(renderEl()).toHaveAttribute('target', '');
+    expect(renderEl({ newTab: true })).toHaveAttribute('target', '_blank');
 
-    expect(getShallow({ target: 'asdf' })).toHaveProp('target', 'asdf');
-    expect(getShallow({ target: 'asdf', newTab: true })).toHaveProp('target', 'asdf');
+    expect(renderEl({ target: 'asdf' })).toHaveAttribute('target', 'asdf');
+    expect(renderEl({ target: 'asdf', newTab: true })).toHaveAttribute('target', 'asdf');
   });
 
   it('sets the target to _blank if the external prop is true and newTab is not false, unless a different target is ' +
       'specified', function() {
-    expect(getShallow({ external: true })).toHaveProp('target', '_blank');
+    expect(renderEl({ external: true })).toHaveAttribute('target', '_blank');
+    expect(renderEl({ external: true, target: 'asdf' })).toHaveAttribute('target', 'asdf');
+
+    expect(renderEl({ external: true, newTab: true })).toHaveAttribute('target', '_blank');
+    expect(renderEl({ external: true, newTab: false })).toHaveAttribute('target', '');
 
-    expect(getShallow({ external: true, target: 'asdf' })).toHaveProp('target', 'asdf');
+    expect(renderEl({ external: true, noReferrer: true, target: 'asdf' })).toHaveAttribute('target', 'asdf');
+  });
 
-    expect(getShallow({ external: true, newTab: true })).toHaveProp('target', '_blank');
-    expect(getShallow({ external: true, newTab: false })).toHaveProp('target', '');
+  it('does not add an icon after the children if external is false', function() {
```

Comment:
As I said above the icon presentation is tested via visual test and testing it at this level is an implementation detail.

Reply:
> As I said above the icon presentation is tested via visual test and testing it at this level is an implementation detail.

Removed here: 70a65dd9c54784f2922ba72a48d7db238e0ecc4e